contract/test/builder/init-orca.js

```diff
@@ -1,39 +1,148 @@
         },
+        chainDetails,
       },
     ],
   });
 };
 
 export default async (homeP, endowments) => {
   const { writeCoreEval } = await makeHelpers(homeP, endowments);
-  await writeCoreEval(startOrcaContract.name, defaultProposalBuilder);
+  const { scriptArgs } = endowments;
+  /** @type {{ values: OrcaOpts }} */
+  const { values: flags } = parseArgs({ args: scriptArgs, options });
+
+  /** @param {string} net */
+  const getNetConfig = net =>
+    fetch(`https://${net}.agoric.net/network-config`)
+      .then(res => res.text())
+      .then(s => JSON.parse(s));
+
+  /** @param {string[]} strs */
+  const parsePeers = strs => {
+    /** @type {[name: string, conn: IBCConnectionID, chan: IBCChannelID, denom:string][]} */
+    // @ts-expect-error XXX ID syntax should be dynamically checked
+    const peerParts = strs.map(s => s.split(':'));
+    const badPeers = peerParts.filter(d => d.length !== 4);
+    if (badPeers.length) {
+      throw Error(
+        `peers must be name:connection-X:channel-Y:denom, not ${badPeers.join(', ')}`,
+      );
+    }
+    return peerParts;
+  };
+
+  /** @type {Record<string, CosmosChainInfo>} */
+  const chainDetails = {};
+
+  if (flags.net) {
+    if (!flags.peer) throw Error('--peer required');
+    /** @type {Record<string, IBCConnectionInfo>} */
+    const connections = {};
+    const portId = 'transfer';
+
+    const { chainName: chainId, rpcAddrs } = await getNetConfig(flags.net);
+    const agd = makeAgd({ execFileSync }).withOpts({ rpcAddrs });
+
+    for (const [peerName, myConn, myChan, denom] of parsePeers(flags.peer)) {
+      console.debug(peerName, { denom });
+      const connInfo = await agd
+        .query(['ibc', 'connection', 'end', myConn])
+        .then(x => x.connection);
+      const { client_id } = connInfo;
+      const clientState = await agd
+        .query(['ibc', 'client', 'state', client_id])
+        .then(x => x.client_state);
+      const { chain_id: peerId } = clientState;
+      console.debug(peerName, { chainId: peerId, denom });
+      chainDetails[peerName] = { chainId: peerId, stakingTokens: [{ denom }] };
+
+      const chan = await agd
+        .query(['ibc', 'channel', 'end', portId, myChan])
+        .then(r => r.channel);
+
+      /** @type {IBCConnectionInfo} */
+      const info = harden({
+        client_id,
+        counterparty: {
+          client_id: connInfo.counterparty.client_id,
+          connection_id: connInfo.counterparty.connection_id,
+          prefix: { key_prefix: 'arbitrary - not used?' },
+        },
+        id: myConn,
+        state: connInfo.state,
+        transferChannel: {
+          channelId: myChan,
+          counterPartyChannelId: chan.counterparty.channel_id,
+          counterPartyPortId: chan.counterparty.port_id,
+          ordering: chan.ordering,
+          portId,
+          state: chan.state,
+          version: chan.version,
+        },
+      });
+      mustMatch(info, IBCConnectionInfoShape);
```

Comment:
p.s. after approval, I added a dynamic pattern, expecting it to fail because this produces `"state": "STATE_OPEN"` instead of `"state": 3`. But our patterns don't seem to care.

It did find a bug were the connection `.id` was `undefined` though, so I'm glad I did it.